src/decorators/providers/component.js

```diff
@@ -96,6 +96,14 @@ export const Component = componentConfig => t => {
 	// to the scope
 	componentWriter.set('bindToController', true, t);
 
+
+	// Must perform some basic shape checking on the config object
+	['inputs', 'providers', 'directives', 'outputs'].forEach(property => {
+		if(config[property] !== undefined && !Array.isArray(config[property])){
+			throw new TypeError(`Component Decorator Error in "${t.name}": Component ${property} must be an array`);
+		}
+	});
+
```

Comment:
I had to move this back up here, because it wasn't working down in the actual module parser.